src/coreclr/nativeaot/Runtime/ThunksMapping.cpp

```diff
@@ -121,6 +121,9 @@ EXTERN_C NATIVEAOT_API void* __cdecl RhAllocateThunksMapping()
         return NULL;
     }
```

Comment:
Should the `PalVirtualProtect`s be ifdefed out for osx-arm64 so that it is symmetric with what we do at the end?
```
#if defined(HOST_OSX) && defined(HOST_ARM64)
    pthread_jit_write_protect_np(...)
#else
   PalVirtualProtect(...)
#endif
```

Reply:
The `PalVirtualProtect` on the data part is actually necessary. I don't remember off-hand the exact reason but it's probably because the data pages are modified from the managed code without `pthread_jit_write_protect_np` guards.

I can change it to skip the other call but I was not sure if it's worth it.

Reply:
I think skipping the other call on osx would look better to make the pre- and post- logic symmetric.

Reply:
I rechecked and actually both `PalVirtualProtect` calls are needed. It's possible to fold the other one into `PalVirtualAlloc` on macOS but it's not necessarily more readable.